docs/architecture/assessment-runtime/configuration.md

```diff
@@ -2,52 +2,40 @@
+:::
 
-Sequence Diagram for updating the database
 ```mermaid
 sequenceDiagram
-    User ->> Configuration: update configuration (component, new control, etc)
-    Configuration ->> DB: updates  
-    DB ->> Configuration: updated
-    jobSpec ->> DB: Queries
-    jobSpec ->> jobSpec: renders job specs  
-    jobSpec ->> DB: updates job specs  
-    DB ->> Configuration: updated
-```
+    participant CM as ConfigurationManager
+    participant AG as ApplicationGateway
+    participant CS as ConfigurationService
+    participant PG as PostgreSQL
+    participant EB as EventBus
 
-Sequence Diagram for the Assessment Runtime to get more jobs
-```mermaid
-sequenceDiagram
-    AssessmentRuntime ->> Configuration: gimme X job specs
-    Configuration ->> DB: get job specs where runtime_id is null limit x
-    DB ->> Configuration: response
-    Configuration ->> AssessmentRuntime: Your Job Specs
-    Configuration ->> DB: update job specs where id = [x] runtime_id = <assessment_id>
-    DB ->> Configuration: response
+    CM->>AG: request_initial_configuration()
+    AG->>CS: forward_request()
+    CS->>PG: query_configuration_data()
+    PG->>CS: return_configuration_data()
+    CS->>AG: respond_with_data(data)
+    AG->>CM: forward_response(data)
+    CM->>CM: update_local_Configuration_Store(data)
 
+    CS->>PG: update_configuration(new_config)
+    PG->>CS: confirm_update()
+    CS->>EB: publish_event("config_change", new_config)
+    EB->>CM: notify_event("config_change")
+    CM->>CM: apply_configuration_change(new_config)
 ```
-Because Users can cause a change to the job specs, we need to have a specific flow for checking if the current job specs are still valid and/or changed. The proposed workflow would be something like this:
-```mermaid
-sequenceDiagram
-    AssessmentRuntime ->> Configuration: gimme my job specs
-    Configuration ->> DB: get job specs where runtime_id is assessment_id
-    DB ->> Configuration: response
-    Configuration ->> AssessmentRuntime: Your Job Specs
-    AssessmentRuntime ->> AssessmentRuntime: delete old job spec from config
 
-```
\ No newline at end of file
+- **Initial Configuration Pull**: The **Configuration Manager (CM)** requests the initial configuration from the **Configuration Service (CS)** through the **Application Gateway (AG)**. The Configuration Service queries the data from the **PostgreSQL (PG)** database and returns it, and the Configuration Manager updates its local Configuration Store.
+
+- **Configuration Update**: When there's a change in the configuration, the Configuration Service updates the PostgreSQL database and confirms the update.
+
+- **Event-Driven Updates**: The Configuration Service then publishes an event to the **Event Bus (EB)** on the "config_change" topic, including the new configuration data. The Event Bus notifies the Configuration Manager, which is listening to the "config_change" topic.
```

Comment:
config changes would be related to assessment actions already owned by a given runtime (either an update on the existing configuration or a deletion of a given event) correct? Are we thinking of a way for "new jobs"?

For me this would be tricky to implement Event-driven, as the main source of information (how many jobs can I have right now) belongs to the runtime.

Reply:
> config changes would be related to assessment actions already owned by a given runtime (either an update on the existing configuration or a deletion of a given event) correct?

Correct. 

So, I guess we're talking about two different ways of adding jobs. One might be [triggered ](https://github.com/compliance-framework/docs/pull/17/files#diff-da9e6f70dbabede632fd7257eada4c3b2de1445b6f82096d8cd06b9f8748a53eR25)by the central configuration by adding new assessments, etc (completely new JobSpecs). The is handled by the event bus, which informs the local configuration manager that the config has been updated in the control plane. This, in turn, lets the Job Manager know about the update. 

> Are we thinking of a way for "new jobs"? Or once a runtime is gets its initial config, no more new jobs are allowed?

This needs to be handled by the Job Manager by requesting more, based on the information it has about its own memory and CPU usage (no change to config but able to handle more load). 

> For me this would be tricky to implement Event-driven, as the main source of information (how many jobs can I have right now) belongs to the runtime.

This is handled by the Job Manager itself by acquiring resource usage information from the Resource Monitor (or by checking it through any other means if we decide to skip developing a Resource Manager). No event bus is involved here, just in-proc calls.

Reply:
Generally speaking we agree, so Approving and Merging - but I think it would eventually also be a call to configuration serivce in the control plane (instead of only in-proc calls), as we need, on the control plane, to do a bookeeping on the "jobsspecs" that are already assigned to runtimes